carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -216,14 +232,14 @@ protected boolean ensureCredentials() {
 
             if (url != null) {
                 try {
-                    amazonCredentials = new PropertiesCredentials(url.openStream());
+					amazonCredentials = getRemoteFileCredentials(url);
```

Comment:
I preferred what you had before. My original change here was just to get a classpath resource (I made the URL change from the original source)